lib/browser/mixins/browser_wait.ts

```diff
@@ -154,7 +172,10 @@ export default abstract class BrowserWait extends BrowserNavigation {
                 return value === null;
             }, timeout, selector);
         } catch (err) {
-            throw new TimeoutError("waitUntilEnabled", `Waiting for element "${selector}" to be enabled`, timeout);
+            if (err instanceof TimeoutError) {
```

Comment:
Same on all these checks, Ideally, the decorator should handle this, if possible